src/Analyzers/CSharp/Analyzers/UsePatternCombinators/CSharpUsePatternCombinatorsDiagnosticAnalyzer.cs

```diff
@@ -128,6 +148,37 @@ private static bool IsTrivial(AnalyzedPattern pattern)
             };
         }
 
+        private static bool IsPure(AnalyzedPattern pattern)
+        {
+            if (pattern is Constant)
+                return true;
+
+            if (pattern.Target is Operations.IPropertyReferenceOperation prop)
+            {
+                return IsAutoProperty(prop.Property);
```

Comment:
i thinkwe  can assume any property is idempotent.  Indeed, that's the assumption the compielr makes when it comes to patterns.  it may only execute the check one time even if called multiple times. plus a property that changes values across many calls woudl be enormously smelly.

Other features also make this assumption, so this seems reasonable tom e.  Indeed, i think i would reverse this and assume everything is safe except for method calls.

Reply:
if you do want thsi, you can just write: `prop.Property.GetBackingFieldIfAny() != null`.